### PR TITLE
Add migration guide from 2.x to 3.x

### PR DESCRIPTION
**Pull request by bchen-twilio**
Migration guide from 2.x to 3.x
- `TVOCallInvite` Changes
- Specifying a media region
- `TVOConnectOptions` & `TVOAcceptOptions`
- Ringing
- Audio Device
- Custom Parameters

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +223,166 @@ There are number of techniques you can use to ensure that access token expiry is
 - Retain the access token along with the timestamp of when it was requested so you can verify ahead of time whether the token has already expired based on the `time-to-live` being used by your server.
 - Prefetch the access token whenever the `UIApplication`, or `UIViewController` associated with an outgoing call is created.
 
+## Migragion Guide
```
Comment (idelgado) [inline README.md]: Migration Guide

File: README.md [hunk 2]
```diff
@@ -222,6 +223,187 @@ There are number of techniques you can use to ensure that access token expiry is
...
+let connectOptions: TVOConnectOptions = TVOConnectOptions(accessToken: accessToken) { (builder) in
+    builder.iceOptions = iceOptions
+}
+
+let acceptOptions: TVOAcceptOptions = TVOAcceptOptions(callInvite: callInvite!) { (builder) in
+    builder.iceOptions = iceOptions
+}
+```
+
+### Audio Device
```
Comment (piyushtank) [inline README.md]: This section needs to be re written. I would devide it in to following -

## Audio Device

### TVIDefaultAudioDevice

### CallKit

### TVIAudioDevice

File: README.md [hunk 3]
```diff
@@ -222,6 +223,187 @@ There are number of techniques you can use to ensure that access token expiry is
 - Retain the access token along with the timestamp of when it was requested so you can verify ahead of time whether the token has already expired based on the `time-to-live` being used by your server.
 - Prefetch the access token whenever the `UIApplication`, or `UIViewController` associated with an outgoing call is created.
 
+## Migration Guide
```
Comment (piyushtank) [inline README.md]: did we have stats of anything similar to stats in 2.x ? should migration guide include `stats` ? @idelgado ?

File: README.md (hunk 2 above)
Comment (piyushtank) [reply README.md]: This section should also talk about deprecation of `CallKit` category from `TwilioVoice`.

File: README.md (hunk 3 above)
Comment (bchen-twilio) [reply README.md]: Nope stats is not available on 2.x. Might not be a bad idea.

Comment (idelgado) [reply README.md]: Yup. Looks like I missed that. In retrospect the migration guide could have excluded new features and become it's own section altogether. For now we can put everything in the migration guide but long-term we can have 2 sections: `Migration Guide` and `New Features`.

File: README.md (hunk 2 above)
Comment (bchen-twilio) [reply README.md]: I think the guide copied from the changelog does cover both default audio device and the TVOAudioDevice protocol and clearly that the `CallKitIntegration` category has been deprecated.

File: README.md [hunk 4]
```diff
@@ -222,6 +223,248 @@ There are number of techniques you can use to ensure that access token expiry is
...
+    builder.iceOptions = iceOptions
+}
+
+let acceptOptions: TVOAcceptOptions = TVOAcceptOptions(callInvite: callInvite!) { (builder) in
+    builder.iceOptions = iceOptions
+}
+```
+
+### Audio Device
+
```
Comment (piyushtank) [inline README.md]: Lets add here (before the start of `TVIDefaultAudioDevice` section), 

### Audio Device

3.X SDK introduces audio device APIs. You should set your audio device on `TwilioVoice.audioDevice` before performing any other actions with the SDK such as creating Tracks or connecting to a Call.

#### TVODefaultAudioDevide
..

**Pull request merged**